README.md

```diff
@@ -21,20 +32,196 @@ Install-Package Microsoft.IO.RecyclableMemoryStream
+
+There are two versions of the large pool: Linear or Exponential.
+
+Linear is the default. You specify a multiple and a maximum size, and an array of buffers, from size (1 * multiple), (2 * multiple), (3 * multiple), ... maximum is created. For example, if you specify a multiple of 1 MB and maximum size of 8 MB,
+then you will have an array of length 8. The first slot will contain 1 MB buffers, the second slot 2 MB buffers, and so on.
+
+Exponential buffers tweaks this slightly. Instead of linearly growing, the buffers double in size for each slot. For example, if you specify a multiple of 256KB, and a maximum size of 8 MB, you will have an array of length 6, the slots containing buffers of size 256KB, 512KB, 1MB, 2MB, 4MB, and 8MB.
+
+![Pool Image Comparison](poolcomparison.png)
+
+Which one should you use? That depends on your usage pattern. If you have an unpredictable large buffer size, perhaps the linear one will be more suitable. If you know that a longer stream length is unlikely, but you may have a lot of streams in the smaller size, picking the Exponential could lead to less overall memory usage (which was the reason this form was added).
+
+Buffers are created, on demand, the first time they are requested and nothing suitable already exists in the pool. After use, these buffers will be returned to the pool through the `RecyclableMemoryStream`'s `Dispose` method. When that return happens, the `RecyclableMemoryStreamManager` will use the properties `MaximumFreeSmallPoolBytes` and `MaximumFreeLargePoolBytes` to determine whether to put those buffers back in the pool, or let them go (and thus be garbage collected). It is through these properties that you determine how large your pool can grow. If you set these to 0, you can have essentially unbounded pool growth, which is essentially indistinguishable from a memory leak. For every application, you must determine through analysis and experimentation the appropriate balance between pool size and garbage collection.
+
+If you forget to call a stream's `Dispose` method, this could cause a memory leak. To help you prevent this, each stream has a finalizer that will be called by the CLR once there are no more references to the stream. This finalizer will raise an event or log a message about the leaked stream.
+
+## Usage
+
+You can jump right in with no fuss by just doing a simple replacement of MemoryStream with something like this:
+
+```
+var sourceBuffer = new byte[]{0,1,2,3,4,5,6,7}; 
```

Comment:
You say on line 90 that it should be declared once and live for the entire process - but I suspect there is some percentage of people who will not understand this.

imo the no-fuss jump-in is something like this (please rename as needed):

```csharp
    public static class Streams
    {
        public static readonly RecyclableMemoryStreamManager Manager = new RecyclableMemoryStreamManager();
    }

    public class MyClass
    {
        public void WriteToStream()
        {
            var sourceBuffer = new byte[] { 0, 1, 2, 3, 4, 5, 6, 7 };
            using (var stream = Streams.Manager.GetStream())
            {
                stream.Write(sourceBuffer, 0, sourceBuffer.Length);
            }
        }
    }
```

Reply:
Good point. Let me redo some of these examples.